src/app.ts

```diff
@@ -90,18 +109,63 @@ export class App extends Construct {
     // the necessary operations. We do however want to preserve the distributed validation.
     validate(this);
 
-    const simpleManifestNamer = (chart: Chart) => `${Names.toDnsLabel(chart)}.k8s.yaml`;
-    const manifestNamer = hasDependantCharts ? (chart: Chart) => `${index.toString().padStart(4, '0')}-${simpleManifestNamer(chart)}` : simpleManifestNamer;
+    const simpleManifestNamer = (construct: Construct) => `${Names.toDnsLabel(construct)}.k8s.yaml`;
 
     const charts: IConstruct[] = new DependencyGraph(Node.of(this)).topology().filter(x => x instanceof Chart);
 
-    let index = 0;
-    for (const node of charts) {
-      const chart: Chart = Chart.of(node);
-      Yaml.save(path.join(this.outdir, manifestNamer(chart)), chartToKube(chart));
-      index++;
+    switch (this.yamlOutputType) {
+      case YamlOutputType.FILE_PER_APP:
+        let apiObjectList:ApiObject[] = [];
+
+        for (const node of charts) {
+          const chart: Chart = Chart.of(node);
+          apiObjectList = apiObjectList.concat(chartToKube(chart));
+        }
+
+        if (apiObjectList.length > 0) {
+          Yaml.save(
+            path.join(this.outdir, 'app.k8s.yaml'), // There is no "app name", so we just hardcode the file name
+            apiObjectList.map((apiObject) => apiObject.toJson()),
+          );
+        }
+        break;
+
+      case YamlOutputType.FILE_PER_CHART:
+        const chartManifestNamer = hasDependantCharts ? (chart: Chart) =>
+          `${index.toString().padStart(4, '0')}-${simpleManifestNamer(chart)}` : simpleManifestNamer;
+
+        let index = 0;
+        for (const node of charts) {
+          const chart: Chart = Chart.of(node);
+          const apiObjects = chartToKube(chart);
+
+          Yaml.save(
+            path.join(this.outdir, chartManifestNamer(chart)),
+            apiObjects.map((apiObject) => apiObject.toJson()),
+          );
+
+          index++;
+        }
+        break;
+
+      case YamlOutputType.FILE_PER_RESOURCE:
+        for (const node of charts) {
+          const chart: Chart = Chart.of(node);
+          const apiObjects = chartToKube(chart);
+
+          apiObjects.forEach((apiObject) => {
+            if (!(apiObject === undefined)) {
+              const fileName = `${`${apiObject.apiVersion.replace(/\//g, '-')}.${apiObject.kind}.${apiObject.metadata.name}`
+                .replace(/[^0-9a-zA-Z-_.]/g, '')}.k8s.yaml`;
```

Comment:
I think this could result in some fairly long file names. e.g. if we have a custom definition of kind "elasticsearches.elasticsearch.k8s.elastic.co" and cdk8s assigns it the name "test-chart-elasticsearch-c819bb71", then the generated file name would be:

```
apiextensions.k8s.io-v1beta1.elasticsearches.elasticsearch.k8s.elastic.co.test-chart-elasticsearch-c819bb71.k8s.yaml
```

Is it possible to shorten this for readability? (Many OS file systems also have a 255 character file limit). Maybe we could omit the api version, or use:

```suggestion
              const fileName = `${Names.toDnsLabel(apiObject)}.k8s.yaml`;
```

Reply:
I think dropping the version should shorten this up enough while providing the detail I was going for